### ISSUE TITLE
Support parallel build steps in pipeline

### ISSUE THREAD

**Issue by KevinGrandon** [buildkite-plugins/docker-compose-buildkite-plugin#60]
We would like to build multiple services through docker-compose across machines to parallelize build time.

Given a pipeline like the following we would expect service 1 and service 2 to be built in parallel. I believe today we can build them, but for some reason further steps will be unable to download the first service and they will be built locally.

```
- name: "SERVICE 1"
  plugins:
    docker-compose:
      build: service1
      image-repository: <repo>
      config: service1/docker-compose.yml
- name: "SERVICE 2"
  plugins:
    docker-compose:
      build: service2
      image-repository: <repo>
      config: service2/docker-compose.yml
- wait
- name: "1st service script"
  command: "ci/script"
  plugins:
    docker-compose:
      run: service1
      config: webapp/docker-compose.yml
- name: "2nd service script"
  command: "ci/script"
  plugins:
    docker-compose:
      run: service2
      config: service2/docker-compose.yml
```

**Comment by toolmantim** [comment]
I've just looked into this, and it seems ever since we switched to using a ['count' meta-data key](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/blob/8a7eb88eee8b5e3bc8516ea2762dc2134c14f9c8/hooks/commands/build.sh#L40) you can't do these things in parallel.

This is definitely a bug we should fix, and I think we can do without the count fairly easily. I'll keep you posted with a fix.

**Comment by lox** [comment]
Had a go at this, unwound a lot of complexity in #62

**Comment by toolmantim** [comment]
This is all fixed in v1.4.0. Give it a shot @KevinGrandon!

**Comment by KevinGrandon** [comment]
This is working great now, thanks for the quick turnaround!
